### PR TITLE
sunxi: convert to multi-variable device title

### PR DESCRIPTION
**Pull request by luaraneda**
Use data from the linux-sunxi community wiki to fill-in missing device information

### CODE REVIEW

**Comment by luaraneda** [comment]
CC maintainer: @wigyori

File: target/linux/sunxi/image/cortex-a7.mk [hunk 1]
```diff
@@ -29,7 +31,9 @@ TARGET_DEVICES += sun7i-a20-olinuxino-lime2
 
 
 define Device/sun7i-a20-olinuxino-lime2-emmc
-  DEVICE_TITLE:=Olimex A20-OLinuXino-LIME2-eMMC
+  DEVICE_VENDOR := Olimex
+  DEVICE_MODEL := A20-OLinuXino-LIME2
+  DEVICE_VARIANT := eMMC
```
Comment (hauke) [inline target/linux/sunxi/image/cortex-a7.mk]: There is a eMMC and a SDCard variant, but the SDCard variant does not have a DEVICE_VARIANT is that ok?

Comment (aparcar) [reply target/linux/sunxi/image/cortex-a7.mk]: I guess it'd be better to add SDCard as variant...

Comment (luaraneda) [reply target/linux/sunxi/image/cortex-a7.mk]: Official product page (for reference): https://www.olimex.com/Products/OLinuXino/A20/A20-OLinuXino-LIME2/open-source-hardware

From what I understood, the new variables should be used to differentiate between hardware variants, not firmware.

The eMMC variant for this board also has an SDCard slot, so in theory it should be able to boot the other image too. In fact, looking at its device tree, it only adds and additional mmc node.

This issue is applicable to all devices that have two or more storage location where OpenWRT could be installed

I think @aparcar and @adrianschmutzler were discussing how to handle this. Have you reached a consensus?

Comment (luaraneda) [reply target/linux/sunxi/image/cortex-a7.mk]: Having said all that. I have no problem on adding SDCard to the other variant. I just want consistency on the names

**Comment by dangowrt** [comment]
Cherry-picked onto master, thanks!

File: target/linux/sunxi/image/cortex-a7.mk (hunk 1 above)
Comment (adrianschmutzler) [reply target/linux/sunxi/image/cortex-a7.mk]: Just to answer the question: From my point of view, we have not reached a clear consensus here, so it is somehow left to the gut feeling of author and reviewer. However, I can live with this being not regulated for now.

**Pull request closed without merging**